lib/ratelimiter.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+var cluster = require('cluster');
+
+var RateLimiter = require('limitation');
+
+/**
+ * Cluster master RateLimiter wrapper
+ */
+function RateLimiterMaster(options) {
+    var self = this;
+    this._limiter = new RateLimiter(options);
+    this._limiter.on('blocks', function(blocks) {
+        self._sendBlocksToWorkers(blocks);
+    });
+}
+
+RateLimiterMaster.prototype._sendBlocksToWorkers = function(blocks) {
+    var workers = cluster.workers;
+    Object.keys(workers).forEach(function(key) {
+        var worker = workers[key];
+        if (worker) {
+            try {
+                worker.send({
+                    type: 'ratelimiter_blocks',
+                    value: blocks,
+                });
+            } catch (e) {}
+        }
+    });
+};
+
+
+/**
+ * Initialize the internal limiter.
+ * @return {Promise<RateLimiter>}
+ */
+RateLimiterMaster.prototype.setup = function() {
+    return this._limiter.setup();
+};
+
+RateLimiterMaster.prototype.updateCounters = function(counters) {
+    var keys = Object.keys(counters);
+    for (var i = 0; i < keys.length; i++) {
+        var key = keys[i];
+        var counter = counters[key];
+        var limits = Object.keys(counter.limits);
+        var minLimit = Math.min.apply(null, limits);
+        this._limiter.isAboveLimit(key, minLimit, counter.value);
+    }
+};
+
+/**
+ * Cluster worker side RateLimiter
+ *
+ * Performs local checks, and communicates with the actual rate limiter in the
+ * master process.
+ */
+function RateLimiterWorker(options) {
+    var self = this;
+    this._interval = options && options.interval || 5000;
+    // Local counters and blocks. Contain objects with `value` and `limits`
+    // properties.
+    this._counters = {};
+    this._blocks = {};
+
+    // Start the periodic send-to-master process in the background.
+    setInterval(function() {
+        self._sendToMaster();
+    }, self._interval);
+}
+
+/**
+ * Send current counters to the master process.
+ */
+RateLimiterWorker.prototype._sendToMaster = function() {
+    var counters = this._counters;
+    this._counters = {};
+    process.send({
```

Comment:
It's possible for a worker to die between enumerating workers & sending the message, and a failing send should not take down the master in that case. However, if the master dies, there isn't much left to do in a worker. It will just exit.